### PR TITLE
Deprecate numeral.js

### PR DESCRIPTION
**Pull request by acmei**
#165 Deprecate numeral.js

* Remove numeral.js library in favor of using d3's built in number formatter (d3-format)
* Deprecate `format` prop from `XAxisLabels` and `YAxisLabels` components in favor of a `formats` prop that receives an array of formats.
* Numerical formatting in `SankeyDiagram` now uses d3-format instead of numeral

### CODE REVIEW

File: src/YAxisLabels.js [hunk 1]
```diff
@@ -143,17 +143,12 @@ class YAxisLabels extends React.Component {
...
-     */
-    format: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
     /**
      * Formats to use for the labels in priority order. XAxisLabels will try to be smart about which format
-     * to use that keeps the labels distinct and provides the least amount of collisions when rendered.
+     * to use that keeps the labels distinct and provides the least amount of collisions when rendered. Formats
+     * can be either string(s) or function(s), utilizing d3-format.
      *
-     * For example, given labels with real numbers one can pass in 0.[0] to round to the first significant digit
+     * For example, given labels with real numbers one can pass in [".0%"] for a rounded percentage, like 12%.
```
Comment (krissalvador27) [inline src/YAxisLabels.js]: Thanks for updating the documentation! Let's also add that we've made the decision to use `d3.format` as our format function :)

File: src/utils/Label.js [hunk 2]
```diff
@@ -5,18 +5,20 @@ import tail from "lodash/tail";
...
+  // Number format examples for a value of 1234.55555
+  // https://github.com/d3/d3-format
   const defaultNumberFormats = [
-    "0.[00]a",
-    "0,0",
-    "0.[0]",
-    "0.[00]",
-    "0.[0000]",
-    "0.[000000]"
+    "~s", // 1.23456k
```
Comment (krissalvador27) [inline src/utils/Label.js]: Thanks for commenting the output, super helpful. Inspired to do this for `defaultTimeFormats` now.

File: src/YAxisLabels.js (hunk 1 above)
Comment (acmei) [reply src/YAxisLabels.js]: This is mentioned in the line above.

File: docs/src/docs/XAxisLabels/propDocs.json [hunk 3]
```diff
@@ -198,18 +184,10 @@
...
-          },
-          {
-            "name": "func"
-          }
-        ]
+        "name": "func"
       },
       "required": false,
-      "description": "Format to use for the labels or accessor that returns the updated label.\n\nFor example, given labels with real numbers one can pass in 0.[0] to round to the first significant digit.",
+      "description": "An accessor function that returns the updated label.\n\nFor example, given labels with real numbers one can pass in the following accessor:\n(label) => `{label}%`.",
```
Comment (krissalvador27) [inline docs/src/docs/XAxisLabels/propDocs.json]: Small nit but just to mirror the formats description can it be 

```
An accessor function that returns the updated label.
For example, given labels with real numbers one can pass in the following accessor:\n(label) => `{label}%` to display percentages.
```

**Pull request merged**